### PR TITLE
 [TAGGING] Support custom tagging for application tier

### PR DESCRIPTION
**Pull request by KimForss**
## Problem
Customers want the ability to apply tags on the Virtual machines at creation time

## Solution
Add a property in the application node in the json that supports providing tags for the application tier VMs

## Tests
Add the fpollowing snippet to the input json
"app_tags" :{
"tagname": "Value",
"tagname2": "Value"
}
"scs_tags" :{
"tagname": "Value",
"tagname2": "Value"
}
"web_tags" :{
"tagname": "Value",
"tagname2": "Value"
}

## Notes
<Additional comments for the PR>

### CODE REVIEW

File: deploy/terraform/terraform-units/modules/sap_system/common_infrastructure/variables_local.tf [hunk 1]
```diff
@@ -129,7 +129,12 @@ locals {
 
   //Enable APP deployment
   enable_app_deployment = try(var.application.enable_deployment, false)
-
+  
+ // Support for tags
+  app_tags = try(var.application.app_tags,[])
+  scs_tags = try(var.application.scs_tags,[])
+  web_tags = try(var.application.web_tags,[])
```
Comment (rukawata) [inline deploy/terraform/terraform-units/modules/sap_system/common_infrastructure/variables_local.tf:136]: default should be {}

Comment (rukawata) [reply deploy/terraform/terraform-units/modules/sap_system/common_infrastructure/variables_local.tf:136]: The tag vars are defined in common_infrastructure, app_tier can't use them.

**Review by rukawata — COMMENTED** [review]
Thanks for the PR. Some problems.

File: deploy/terraform/terraform-units/modules/sap_system/common_infrastructure/variables_local.tf (hunk 1 above)
Comment (KimForss) [reply deploy/terraform/terraform-units/modules/sap_system/common_infrastructure/variables_local.tf]: doh :)

File: deploy/terraform/terraform-units/modules/sap_system/app_tier/variables_local.tf [hunk 2]
```diff
@@ -199,6 +199,11 @@ locals {
     "version"         = try(var.application.scs_os.version, local.scs_custom_image ? "" : local.app_os.version)
   }
 
+  app_tags = try(var.application.app_tags,{})
+  scs_tags = try(var.application.scs_tags,{})
+  web_tags = try(var.application.web_tags,{})
+
+
```
Comment (rukawata) [inline deploy/terraform/terraform-units/modules/sap_system/app_tier/variables_local.tf:206]: nit:
```suggestion

```

File: deploy/terraform/terraform-units/modules/sap_system/app_tier/variables_local.tf [hunk 3]
```diff
@@ -199,6 +199,11 @@ locals {
     "version"         = try(var.application.scs_os.version, local.scs_custom_image ? "" : local.app_os.version)
   }
 
+  app_tags = try(var.application.app_tags,{})
```
Comment (rukawata) [inline deploy/terraform/terraform-units/modules/sap_system/app_tier/variables_local.tf:202]: Please add a comment like in the PR for db VMs.

File: deploy/terraform/terraform-units/modules/sap_system/common_infrastructure/variables_local.tf [hunk 4]
```diff
@@ -129,8 +129,8 @@ locals {
 
   //Enable APP deployment
   enable_app_deployment = try(var.application.enable_deployment, false)
-
-  //Enable SID deployment
+  
+   //Enable SID deployment
```
Comment (rukawata) [inline deploy/terraform/terraform-units/modules/sap_system/common_infrastructure/variables_local.tf:133]: please revert this change...

**Review by rukawata — COMMENTED** [review]
Most looks fine! Just make sure you rebase:
```
  ~ os_disk {
          ~ caching                   = "None" -> "ReadWrite"
            disk_size_gb              = 30
            name                      = "U6068-EAUS-SAP-PRD_prdapp00le81-OsDisk"
            storage_account_type      = "Standard_LRS"
            write_accelerator_enabled = false
        }
```

**Comment by KimForss** [comment]
Replaced with a new
